drivers/gpio/gpio_sx1509b.c

```diff
@@ -104,112 +106,107 @@ static int gpio_sx1509b_config(struct device *dev, int access_op, u32_t pin,
 	const struct gpio_sx1509b_config *cfg = dev->config->config_info;
 	struct gpio_sx1509b_drv_data *drv_data = dev->driver_data;
 	struct gpio_sx1509b_pin_state *pins = &drv_data->pin_state;
+	struct {
+		u8_t reg;
+		struct gpio_sx1509b_pin_state pins;
+	} __packed outbuf;
 	int ret = 0;
+	u16_t bit = BIT(pin);
```

Comment:
Have I missed something or would replacing it with the following:
```
u16_t bits = ((access_op == GPIO_ACCESS_BY_PORT) ? 0xFFFF : BIT(pin));
```
allow you to keep the ability to configure entire port at once (although such feature does not seem to have many use cases, indeed)?

Reply:
In fact configuring the entire port at once is very important, it just can't be done within the existing Zephyr GPIO API, which apparently is designed to configure every pin on the port to the same configuration.  Which is indeed useless in most cases and motivated removing the port-level API from this driver in an earlier commit so the rest of this would be simpler.

I've got a more significant update to the SX1509B driver pending, that includes configuring all pins in one go (used in nrf52_pca20020 startup), changing multiple pin values in one transaction, and LED driver support.  It's part of what's waiting for this to be merged.